src/foam/nanos/approval/ApprovalRequest.js

```diff
@@ -0,0 +1,440 @@
+/**
+ * @license
+ * Copyright 2020 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+ foam.CLASS({
+  package: 'foam.nanos.approval',
+  name: 'ApprovalRequest',
+  documentation: 'Approval requests are stored in approvalRequestDAO and' +
+  'represent a single approval request for a single user.',
+
+  implements: [
+    'foam.nanos.auth.CreatedAware',
+    'foam.nanos.auth.CreatedByAware',
+    'foam.nanos.auth.LastModifiedAware',
+    'foam.nanos.auth.LastModifiedByAware'
+  ],
+
+  javaImports: [
+    'foam.core.FObject',
+    'foam.core.X',
+    'foam.dao.DAO',
+    'foam.nanos.logger.Logger'
+  ],
+
+  requires: [
+    'foam.dao.AbstractDAO',
+    'foam.nanos.approval.ApprovalStatus'
+  ],
+
+  imports: [
+    'approvalRequestDAO',
+    'ctrl'
+  ],
+
+  tableColumns: [
+    'id',
+    'description',
+    'classification',
+    'objId',
+    'viewReference',
+    'approver',
+    'status',
+    'memo',
+    'approve',
+    'reject'
+  ],
+
+  sections: [
+    {
+      name: 'basicInformation',
+      isAvailable: () => false
+    },
+    {
+      name: 'requestDetails'
+    },
+    {
+      name: 'supportDetails'
+    },
+    {
+      name: '_defaultSection',
+      isAvailable: () => false
+    }
+  ],
+
+  properties: [
+    {
+      class: 'Long',
+      name: 'id',
+      section: '_defaultSection',
+      visibility: 'RO',
+      documentation: 'Sequence number.'
+    },
+    {
+      class: 'Reference',
+      of: 'foam.nanos.auth.User',
+      name: 'approver',
+      section: 'requestDetails',
+      documentation: `The user that is requested for approval. When set, "group" property is ignored.`,
+      tableCellFormatter: function(approver) {
+        let self = this;
+        this.__subSubContext__.userDAO.find(approver).then((user)=> {
+          if ( user && user.group ) {
+            self.add(user.group);
+          } else {
+            self.add(approver);
+          }
+        });
+      },
+      visibility: function(approver) {
+        return approver ?
+          foam.u2.DisplayMode.RO :
+          foam.u2.DisplayMode.HIDDEN;
+      },
+    },
+    {
+      class: 'Object',
+      javaType: 'Object',
+      name: 'objId',
```

Comment:
As mentioned in the Approvable above, it would be nice to have access to these in a permissioned/admin section.

Reply:
Was last minute fix, just changed it to permissionRequired: true

Reply:
Was last minute fix, just changed it to permissionRequired: true